data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -118,18 +118,34 @@ private AcknowledgementSet createAcknowledgementSet(Map<TopicPartition, Range<Lo
                 if (result == true) {
                     positiveAcknowledgementSetCounter.increment();
                     offsets.forEach((partition, offsetRange) -> {
-                        int partitionId = partition.partition();
-                        if (!partitionCommitTrackerMap.containsKey(partitionId)) {
-                            OffsetAndMetadata committedOffsetAndMetadata = consumer.committed(partition);
-                            Long committedOffset = Objects.nonNull(committedOffsetAndMetadata) ? committedOffsetAndMetadata.offset() : null;
-
-                            partitionCommitTrackerMap.put(partitionId, new TopicPartitionCommitTracker(partition, committedOffset));
+                        try {
+                            int partitionId = partition.partition();
+                            if (!partitionCommitTrackerMap.containsKey(partitionId)) {
+                                OffsetAndMetadata committedOffsetAndMetadata = null;
+                                synchronized(consumer) {
+                                    committedOffsetAndMetadata = consumer.committed(partition);
+                                }
+                                Long committedOffset = Objects.nonNull(committedOffsetAndMetadata) ? committedOffsetAndMetadata.offset() : null;
+                                partitionCommitTrackerMap.put(partitionId, new TopicPartitionCommitTracker(partition, committedOffset));
+                            }
+                            OffsetAndMetadata offsetAndMetadata = partitionCommitTrackerMap.get(partitionId).addCompletedOffsets(offsetRange);
+                            updateOffsetsToCommit(partition, offsetAndMetadata);
+                        } catch (Exception e) {
+                            LOG.error("Failed to seek to last committed offset upon positive acknowledgement "+partition, e);
                         }
-                        OffsetAndMetadata offsetAndMetadata = partitionCommitTrackerMap.get(partitionId).addCompletedOffsets(offsetRange);
-                        updateOffsetsToCommit(partition, offsetAndMetadata);
                     });
                 } else {
-                    positiveAcknowledgementSetCounter.increment();
+                    negativeAcknowledgementSetCounter.increment();
+                    offsets.forEach((partition, offsetRange) -> {
+                        try {
+                            synchronized(consumer) {
+                                OffsetAndMetadata committedOffsetAndMetadata = consumer.committed(partition);
+                                consumer.seek(partition, committedOffsetAndMetadata);
+                            }
+                        } catch (Exception e) {
+                            LOG.error("Failed to seek to last committed offset upon negative acknowledgement "+partition, e);
```

Comment:
Please use Log4j's string interpolation here. See comment above.

Reply:
Will do this in the next PR.